utils/sites.js

```diff
@@ -24,60 +23,121 @@ const submitParticipantsData = async (req, res, site) => {
         if(!authorized){
             return res.status(401).json(getResponseJSON('Authorization failed!', 401));
         }
+
         siteCode = authorized.siteCode;
     }
 
-    if(req.body.data === undefined) return res.status(400).json(getResponseJSON('data is undefined in request body.', 400));
-    if(req.body.data.length === undefined || req.body.data.length < 1) return res.status(400).json(getResponseJSON('data array doesn\'t have any element.', 400));
+    if(req.body.data === undefined) return res.status(400).json(getResponseJSON('Bad request. Data is not defined in request body.', 400));
+    if(!Array.isArray(req.body.data)) return res.status(400).json(getResponseJSON('Bad request. Data must be an array.', 400));
+    if(req.body.data.length === undefined || req.body.data.length < 1) return res.status(400).json(getResponseJSON('Bad request. Data array does not have any elements.', 400));
```

Comment:
`if (req.body.data.length === 0)`

Reply:
would this capture the scenario that length is undefined? or are you just suggesting I replace "req.body.data.length < 1" with "req.body.data.length === 0" ?

Reply:
Great point, thank you for pointing this out 👍